pkg/ottl/functions.go

```diff
@@ -381,25 +381,25 @@ type optionalManager interface {
 }
 
 type Optional[T any] struct {
-	val      T
-	hasValue bool
+	Val      T
+	HasValue bool
```

Comment:
We should not need to export these values.

Reply:
I've reverted this change.